src/main/java/com/rabbitmq/jms/admin/RMQConnectionFactory.java

```diff
@@ -1026,6 +1037,32 @@ public void setAmqpConnectionFactoryPostProcessor(Consumer<com.rabbitmq.client.C
         this.amqpConnectionFactoryPostProcessor = amqpConnectionFactoryPostProcessor;
     }
 
+
+    /**
+     * Sets a post-processor for the created {@link com.rabbitmq.client.Connection}.
+     * <p>
+     * The post-processor is called after the {@link com.rabbitmq.client.Connection} is created and established. This callback
+     * can be used to customize the {@link com.rabbitmq.client.Connection} e.g. adding a {@link com.rabbitmq.client.BlockedListener}
+     * that emits a log message when this connection gets blocked:
+     * <pre>
+     * {@code
+     * RMQConnectionFactory factory = new RMQConnectionFactory();
+     * // ...
+     * factory.setAmqpConnectionPostProcessor(connection ->
+     *                 connection.addBlockedListener(
+     *                         reason -> log.warn("Connection blocked: {}", reason),
+     *                         () -> log.info("Connection unblocked"))
+     *         );
+     * }
+     * </pre>
+     *
+     * @param amqpConnectionPostProcessor callback that processes the AMQP connections after they are established
+     * @since 2.10.0
+     */
+    public void setAmqpConnectionPostProcessor(Consumer<com.rabbitmq.client.Connection> amqpConnectionPostProcessor) {
```

Comment:
Unless "post processor" is a term widely used in the JMS world, we could use `configure` for the name, just like we do in the Java client for socket "post-configuration" https://github.com/rabbitmq/rabbitmq-java-client/blob/b5396357ad80d439ff6cdfae99473cb3ce5c3086/src/main/java/com/rabbitmq/client/impl/SocketFrameHandlerFactory.java#L59.

That said, I do not feel too strongly about this.

Reply:
> Unless "post processor" is a term widely used in the JMS world

I don't know that - but it sounds very "Spring Boot-like" for sure 😁 

Having said that I don't feel strongly about this either. 

The main reason I chose "post processor" is because of consistency - the callback for the `ConnectionFactory` is called [`amqpConnectionFactoryPostProcessor`](https://github.com/rabbitmq/rabbitmq-jms-client/blob/main/src/main/java/com/rabbitmq/jms/admin/RMQConnectionFactory.java#L164)

Reply:
That is right, it sounds "Spring-y" and it is not widely used in the JMS world. I do not feel strongly about `PostProcessor`, but I think we should stick to it for consistency's sake because there is already the `amqpConnectionFactoryPostProcessor`.

Reply:
If it sounds "Spring-y" then I have no objections :)